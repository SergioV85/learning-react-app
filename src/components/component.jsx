import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}