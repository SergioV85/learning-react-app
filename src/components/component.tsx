import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}