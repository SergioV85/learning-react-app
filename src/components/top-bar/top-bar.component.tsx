import * as React from 'react';

interface ITopbarProps {
  showBackButton?: boolean;
  onBackToList(): void;
}
// tslint:disable:max-line-length
export const Topbar = (props: ITopbarProps) =>
  <div className='c-topbar'>
    <span className='c-topbar__logo'>netflixroulette</span>
    {props.showBackButton ? <button className='btn c-topbar__back-button' onClick={props.onBackToList}> Search </button> : null}
  </div>;
// tslint:enable:max-line-length
