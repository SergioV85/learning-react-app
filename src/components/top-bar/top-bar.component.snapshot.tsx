import { shallow } from 'enzyme';
import * as React from 'react';
import { Topbar } from './top-bar.component';

describe('Topbar Component Snapshot', () => {
  const mockedBackToList = jest.fn();

  it('should renders correctly', () => {
    const topbarSnapshot = shallow(
      <Topbar showBackButton={true} onBackToList={mockedBackToList} />,
    );
    expect(topbarSnapshot)
      .toMatchSnapshot();
  });
});
