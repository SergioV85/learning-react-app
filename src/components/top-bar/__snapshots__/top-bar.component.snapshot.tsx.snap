// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Component Snapshot should renders correctly 1`] = `
<div
  className="c-topbar"
>
  <span
    className="c-topbar__logo"
  >
    netflixroulette
  </span>
  <button
    className="btn c-topbar__back-button"
    onClick={[MockFunction]}
  >
     Search 
  </button>
</div>
`;
